agenta-cli/agenta/sdk/agenta_decorator.py

```diff
@@ -152,6 +163,47 @@ def ingest_files(
             func_params[name] = ingest_file(func_params[name])
 
 
+async def prepare_llm_params_and_begin_tracing(
+    params: Dict[str, Any], result: Union[str, dict]
+) -> None:
+    """Prepares LLM parameters and begins tracing.
+
+    Args:
+        params (Dict[str, Any]): the LLM app parameters
+        result (Union[str, dict]): the output of the llm run
+    """
+
+    def get_prompt_system(config_params: Dict[str, Any]) -> Union[str, None]:
```

Comment:
Oh, I did that based on the number of examples we have. Will there be a prompt that defines the context for the LLM to generate a response from? If you look at the content of the `get_prompt_system`, I am defining three kinds of prompts that would mostly likely serve as the system prompt: `base_prompt`, `system_prompt`, `template_prompt`.

If none of this are defined in the LLM app, then the function returns None.

Reply:
I have written more details about what is expected to be logged in the document. Please check it out.

But overall, we should be agnostic to the name of the configuration. It could be simply that the user does not call the base_prompt as such. So what we need to log in the trace is all the configuration. (I added a comment in the document)